app/data/tasks/deduplicate_records.py

```diff
@@ -0,0 +1,187 @@
+from django.conf import settings
+from django.core.paginator import Paginator
+from django.db.models import Q
+
+from celery import shared_task
+from celery.app.task import Task
+from celery.utils.log import get_task_logger
+
+from data.models import (DedupeJob, RecordDuplicate)
+from ashlar.models import Record
+
+import datetime
+from pytz import timezone
+
+logger = get_task_logger(__name__)
+
+
+class DedupeTask(Task):
+    """Base class for the deduplication task to handle updating status in database"""
+
+    def on_success(self, retval, task_id, args, kwargs):
+        job = DedupeJob.objects.get(celery_task=task_id)
+        job.status = DedupeJob.Status.SUCCESS
+
+    def on_failure(self, exc, task_id, args, kwargs, einfo):
+        """ On failure, delete duplicates found in the unsuccessful run"""
+        job = DedupeJob.objects.get(celery_task=task_id)
+        job.status = DedupeJob.Status.ERROR
+        RecordDuplicate.objects.filter(job=job).delete()
+
+
+def calculate_similarity_score(record1, record2, time_allowance, distance_allowance):
+    """
+    Set a score representing the likelyhood the two records are duplicated, based
+    on the criteria.  score is a float between 0 and 1
```

Comment:
It would be good to note that the records need to be separated by no more than `time_allowance` and `distance_allowance` -- if the separation is more than that, this may output values below 0.